Netkan/Program.cs

```diff
@@ -0,0 +1,175 @@
+using CommandLine;
+using log4net;
+using log4net.Config;
+using log4net.Core;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace CKAN.NetKAN
+{
+    public static class Program
+    {
+        private const int ExitOk = 0;
+        private const int ExitBadOpt = 1;
+        private const int ExitError = 2;
+
+        private static readonly ILog Log = LogManager.GetLogger(typeof(Program));
+
+        private static CmdLineOptions Options { get; set; }
+
+        public static int Main(string[] args)
+        {
+            try
+            {
+                ProcessArgs(args);
+
+                if (Options.File != null)
+                {
+                    Log.InfoFormat("Transforming {0}", Options.File);
+
+                    var moduleService = new ModuleService();
+                    var fileService = new FileService();
+
+                    using (var http = new CachingHttpService(FindCache(new KSPManager(new ConsoleUser(false)))))
+                    {
+                        var netkan = ReadNetkan();
+                        Log.Info("Finished reading input");
+
+                        new NetkanValidator().Validate(netkan);
+                        Log.Info("Input successfully passed pre-validation");
+
+                        var transformer = new NetkanTransformer(
+                            http,
+                            fileService,
+                            moduleService,
+                            Options.GitHubToken,
+                            Options.PreRelease
+                        );
+
+                        var ckan = transformer.Transform(netkan);
+                        Log.Info("Finished transformation");
+
+                        new CkanValidator(netkan, http, moduleService).Validate(ckan);
+                        Log.Info("Output successfully passed post-validation");
+
+                        WriteCkan(ckan);
+                    }
+                }
+                else
+                {
+                    Log.Fatal(
+                        "Usage: netkan [--verbose|--debug] [--debugger] [--prerelease] [--outputdir=...] <filename>"
+                    );
+
+                    return ExitBadOpt;
+                }
+            }
+            catch (Exception e)
+            {
+                Log.Fatal(e.Message);
+
+                if (Options == null || Options.Debug)
+                {
+                    Log.Fatal(e.StackTrace);
+                }
+
+                return ExitError;
+            }
+
+            return ExitOk;
+        }
+
+        private static void ProcessArgs(string[] args)
+        {
+            if (args.Any(i => i == "--debugger"))
+            {
+                Debugger.Launch();
+            }
+
+            Options = new CmdLineOptions();
+            Parser.Default.ParseArgumentsStrict(args, Options);
+
+            BasicConfigurator.Configure();
+            LogManager.GetRepository().Threshold = Level.Error;
```

Comment:
Our default should probably be `Level.Warn` rather than `Level.Error`, but that's the smallest of nits.